### PR TITLE
use mime::Mime for mime types / content types

### PR DESCRIPTION
**Pull request by syphar**

### CODE REVIEW

File: src/storage/mod.rs [hunk 1]
```diff
@@ -801,24 +803,26 @@ impl std::fmt::Debug for Storage {
...
         "text/plain" | "text/troff" | "text/x-markdown" | "text/x-rust" | "text/x-toml" => {
             match file_path.as_ref().extension().and_then(OsStr::to_str) {
-                Some("md") => "text/markdown",
-                Some("rs") => "text/rust",
-                Some("markdown") => "text/markdown",
-                Some("css") => "text/css",
-                Some("toml") => "text/toml",
-                Some("js") => "text/javascript",
-                Some("json") => "application/json",
+                Some("md") => mimes::TEXT_MARKDOWN.clone(),
```
Comment (GuillaumeGomez) [inline src/storage/mod.rs:814]: Instead of cloning, wouldn't be better to use `to_string()` directly?

Comment (syphar) [reply src/storage/mod.rs:814]: I wanted `detect_mime` to return `Mime`. Of course, `&'static Mime` would be better, but `mime_guess` just returns `Mime`. 

In this case, I don't see this really as a performance issue, and having the types / type check here is (to me) more important.

Comment (GuillaumeGomez) [reply src/storage/mod.rs:814]: Hum... Ok for now let's ignore it. It's just that the `Mime` type is actually quite big so it's not great to clone but we can come back to it later.

File: src/storage/mod.rs [hunk 2]
```diff
@@ -801,24 +803,26 @@ impl std::fmt::Debug for Storage {
     }
 }
 
-fn detect_mime(file_path: impl AsRef<Path>) -> &'static str {
+fn detect_mime(file_path: impl AsRef<Path>) -> Mime {
     let mime = mime_guess::from_path(file_path.as_ref())
```
Comment (GuillaumeGomez) [inline src/storage/mod.rs:807]: Can this `mime` creation be moved in the `_ =>` pattern instead of being created unconditionally?

Comment (syphar) [reply src/storage/mod.rs:807]: no, because the first `match` is actually on `Mime.as_str` and not on `file_path`

Comment (GuillaumeGomez) [reply src/storage/mod.rs:807]: Oh I see. Missed that part, sorry.

**Review by GuillaumeGomez — APPROVED** [review]

**Pull request merged**